src/test/java/core/basesyntax/service/implemantation/TransactionExecutorImplTest.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.service.implemantation;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.handlers.BalanceHandler;
+import core.basesyntax.handlers.OperationHandler;
+import core.basesyntax.handlers.PurchaseHandler;
+import core.basesyntax.handlers.ReturnHandler;
+import core.basesyntax.handlers.SupplyHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.TransactionExecutor;
+import core.basesyntax.storage.Storage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class TransactionExecutorImplTest {
+    private static final String FRUIT_NAME = "banana";
+    private static final int FRUIT_QUANTITY = 50;
+    private static final int COUNT_FRUIT = 10;
+    private static final int EXPECTED_FRUIT_QUANTITY = 60;
+    private static final Map<Operation, OperationHandler> operationHandlerMap = Map.of(
+            Operation.BALANCE, new BalanceHandler(),
+            Operation.SUPPLY, new SupplyHandler(),
+            Operation.PURCHASE, new PurchaseHandler(),
+            Operation.RETURN, new ReturnHandler());
+
+    private TransactionExecutor transactionExecutor;
+
+    @BeforeEach
+    void setUp() {
+        transactionExecutor = new TransactionExecutorImpl();
+    }
+
+    @Test
+    void processDate_correctExecutor_ok() {
```

Comment:
You don't need to test your strategy methods in this service, 1 operation just to check if service calls handlers correctly is enough.
![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/82539553/3c92ae5a-d7b8-49bc-a8d9-5c64253df31f)